Blog: Siemens Iot2050 (#1738)
FlowFuse/website: src/blog/2024/03/flowfuse-on-siemens-iot2050.md

---
title: Run FlowFuse on Siemens IoT2050
subtitle: How to install FlowFuse Device Agent on Siemens IoT2050
description: In this guide, we will discuss how to install FlowFuse Device agent on Siemens IoT2050.
date: 2024-03-04
authors: ["grey-dziuba"]
image: 
tags:
    - posts
    - flowfuse
---

> Siemens [announced](https://press.siemens.com/global/en/pressrelease/new-siemens-gateway-between-cloud-company-it-and-production) the IoT2000 series in March of 2020.  With this tool many have been using it to function as a gateway between their plant operations and cloud infrastructure.  Onboard it came with Node-RED pre installed.  In this guide we will show you how to install the FlowFuse device agent.  

Reviewer: ```suggestion
Siemens [announced](https://press.siemens.com/global/en/pressrelease/new-siemens-gateway-between-cloud-company-it-and-production) the IoT2000 series in March of 2020.  With this tool many have been using it to function as a gateway between their plant operations and cloud infrastructure.  Onboard it came with Node-RED pre installed. To manage Node-RED as organisation the FlowFuse agent is recommended, this guide shows you how to do so.
```

<!--more-->

> <div style="background-color: #fff4b9; border:1px solid #ffc400; color: #a27110; padding: 12px; border-radius: 6px; font-style: italic;">Warning: Later in the guide we will be updating Node.js. This will break mraa library. This will prevent communication to the I/O of the device.</div>

Author: IO is used synonomosly through automation and controls industry.  For example if  google "Siemens input output module" the first non-sponsored result from Siemens is, "I/O modules - Industrial Mall"

Is it a question of did I use it right or that I didn't explain what I/O means?

> <div style="background-color: #fff4b9; border:1px solid #ffc400; color: #a27110; padding: 12px; border-radius: 6px; font-style: italic;">Warning: Later in the guide we will be updating Node.js. This will break mraa library. This will prevent communication to the I/O of the device.</div>

Author: Changed to GPIO to match how Siemens referred to it in their forum that is linked.

## Guide

1. First we need to run the standard updates.

```shell
apt-get update
apt-get upgrade
```

2. If you need to migrate your existing Node-RED follow these [instruction](https://flowfuse.com/docs/migration) to backup your existing progress.  From there we will need to remove the existing service that autostart Node-RED by running the following command and reboot:

```shell
systemctl disable node-red.service
reboot -h now
```

3. Confirm that your Node-RED instance is no longer running.

```shell
systemctl status node-red
```

4. Now it is time to upgrade your Node.js version.  To check the version before we get started run ```node -v```.  Then install a tool that will allow you to change your versions of Node.js with the following command.

> <div style="background-color: #fff4b9; border:1px solid #ffc400; color: #a27110; padding: 12px; border-radius: 6px; font-style: italic;">Warning: updating Node.js will break mraa library. This will prevent communication to the I/O of the device.</div>

Reviewer: Can we link to an external discussion to track to see if it gets updated to work?

Author: ```suggestion
<div style="background-color: #fff4b9; border:1px solid #ffc400; color: #a27110; padding: 12px; border-radius: 6px; font-style: italic;">Warning: updating Node.js will break mraa library. This will prevent communication to the I/O of the device. Details can be found <a href="https://support.industry.siemens.com/forum/WW/en/posts/iot2050-node-js-versions/297170">here</a>.</div>
```

```shell

> npm install n -g

Reviewer: Why use `n`, and not `apt-get`?

Author: n is great for switching between versions.  So if you want to switch back it is easier.  It is similar python env.

Reviewer: But you've used apt get, now you introduce another none default tool.

Reviewer: Wonder if it's worth it for such a simple task of upgrading a dependency

Author: If they don't read all the warning going back could be a pain for them.  I would also need to test and see if it didn't break anything.  Reinstall the base os and go through all of the other steps.  I can do that if necessary.  Reduces by 1 step if there are no errors.

Author: https://stackoverflow.com/questions/76318653/how-can-i-install-node-js-version-18-on-ubuntu-22-04

Installing a specific version of Node.JS can sometimes be difficult.
